src/contexts/BlockContext/Provider.tsx

```diff
@@ -1,35 +1,74 @@
 import useAPI from "hooks/useAPI";
 import React, { useCallback, useEffect, useState } from "react";
-import { IGetBlockchainStatusResult } from "types/NXTAPI";
+import { IBlock, IGetBlockchainStatusResult, IGetBlocksResult } from "types/NXTAPI";
+import { CalculateAvgBlocktime } from "utils/common/AvgBlockTime";
+import { BlockPollingFrequency, DefaultBlockFetchQty, DefaultBlockOffset } from "utils/common/constants";
+import { TxCount } from "utils/common/DailyTransactionCount";
 import Context from "./Context";
 
 const BlockProvider: React.FC = ({ children }) => {
   const [blockHeight, setBlockHeight] = useState<number>();
-  const { getBlockchainStatus } = useAPI();
+  const [recentBlocks, setRecentBlocks] = useState<Array<IBlock>>();
+  const [avgBlockTime, setAvgBlockTime] = useState<number>();
+  const [dailyTxs, setDailyTxs] = useState<number>();
+
+  const { getBlockchainStatus, getBlocks } = useAPI();
+
   const fetchBlockHeight = useCallback(async () => {
     if (getBlockchainStatus === undefined) {
-      console.log("returning early, oops");
       return;
     }
 
     const result: false | IGetBlockchainStatusResult = await getBlockchainStatus();
     if (result) {
-      setBlockHeight(result.numberOfBlocks);
+      setBlockHeight(result.numberOfBlocks - 1); // blocks are index'd at 0 so current height is number of blocks minus one
     }
   }, [getBlockchainStatus]);
 
+  const handleFetchRecentBlocks = useCallback(
+    async (first: number, last: number) => {
+      if (getBlocks === undefined) {
+        return;
+      }
+
+      const result: false | IGetBlocksResult = await getBlocks(first, last);
+
+      if (result) {
+        setRecentBlocks(result.blocks);
+      }
+    },
+    [getBlocks]
+  );
+
+  useEffect(() => {
+    handleFetchRecentBlocks(DefaultBlockOffset, DefaultBlockFetchQty); // fetching is done in reverse order so index 0 is the highest block
+  }, [blockHeight, handleFetchRecentBlocks]);
+
   useEffect(() => {
     const timerId = setInterval(() => {
       fetchBlockHeight();
-    }, 5000);
+    }, BlockPollingFrequency);
 
     return () => clearInterval(timerId);
   }, [fetchBlockHeight]);
 
+  useEffect(() => {
+    // TODO: Tooltip explaining how many blocks are avg'd?
```

Comment:
tooltip is a good idea but you would have to put that code into where you display the blocks data (possibly multiple areas) instead of this context provider. still a great note to have here to be sure it gets done!